service/consul/datacenterWatch.go

```diff
@@ -0,0 +1,109 @@
+package consul
+
+import (
+	"errors"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/service"
+)
+
+//DatacenterWatcher checks if datacenters have been updated, based on an interval
+//If they have, then the DatacenterWatcher will update the Instancers dictionary
+type DatacenterWatcher struct {
+	watchInterval time.Duration
+	logger        log.Logger
+	shutdown      chan struct{}
+	environment   Environment
+	options       Options
+}
+
+var (
+	defaultLogger = log.NewNopLogger()
+)
+
+func newDatacenterWatcher(logger log.Logger, environment Environment, options Options) (*DatacenterWatcher, error) {
+	if options.DatacenterWatchInterval == 0 {
+		return nil, errors.New("interval cannot be 0")
+	}
+
+	if logger == nil {
+		logger = defaultLogger
+	}
+
+	return &DatacenterWatcher{
+		watchInterval: options.DatacenterWatchInterval,
+		logger:        logger,
+		shutdown:      make(chan struct{}),
+		environment:   environment,
+		options:       options,
+	}, nil
+
+}
+
+func (i *DatacenterWatcher) Start() {
+	go i.watchDatacenters()
+}
+
+func (i *DatacenterWatcher) Stop() {
+	close(i.shutdown)
+}
+
+func (i *DatacenterWatcher) watchDatacenters() {
+
+	environment := i.environment
+	client := i.environment.Client()
+	logger := i.logger
+	options := i.options
+
+	checkDatacenters := time.NewTicker(i.watchInterval)
+
+	for {
+		select {
+		case <-i.shutdown:
+			return
+		case <-checkDatacenters.C:
+			currentInstancers := environment.Instancers()
+			keys := make(map[string]bool)
+			instancersToAdd := make(service.Instancers)
+
+			datacenters, err := getDatacenters(logger, client, options)
+
+			if err != nil {
+				continue
```

Comment:
Do we need to log an error here? what happens if we always fail to get datacenters?

Reply:
Ah so my reasoning for adding a log here because if the `getDatacenters` errors, it already logs that inside that function, so I thought an additional log wasn't needed here. Would it be better to add a log here so that we know exactly what is calling `getDatacenters` if it errors?